tmd/Topology/analysis.py

```diff
@@ -190,6 +190,25 @@ def histogram_stepped(ph1):
     return bins, results
 
 
+def barcode_bin_centers(ph, num_bins=100, min_bin=None, max_bin=None):
+    """Calculate how many barcode lines are found in each bin.
+       Returns the bin centers and the number of bars that fall within a bin.
+    """
+    ph_2D = [p[:2] for p in ph]  # simplify to ensure ph corresponds to 2d barcode
+    if min_bin is None:
+        min_bin = np.min(ph_2D)
+    if max_bin is None:
+        max_bin = np.max(ph_2D)
+    bins = np.linspace(min_bin, max_bin, num_bins, dtype=float)
+    bin_centers = 0.5 * (bins[:-1] + bins[1:])
+    data = []
+    for i in range(num_bins -1):
+        for p in ph1:
+            if np.max(p) > bins[i+1] and np.min(p) < bins[i]:
+                data.append(bin_centers[i])
```

Comment:
Would that work? 

`[bin_centers[i] for p in ph1 for i in range(num_bins - 1) if np.max(p) > bins[i+1] and np.min(p) < bins[i]]`

Reply:
Yep, almost (ph1 -> ph_2D) ;)

Reply:
Ah this is your change of notation haha

Reply:
why not use `np.histogram`?

Reply:
Because the function kde = stats.gaussian_kde(data) wants data not histrogram values

Reply:
Ah, sorry, I meant `np.digitize`

Reply:
I didn't know that one, it seems to be doing the job indeed! Thanks!